src/cascadia/TerminalSettingsEditor/ColorSchemeViewModel.cpp

```diff
@@ -48,6 +50,25 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         return _Name;
     }
 
+    bool ColorSchemeViewModel::IsDefaultScheme()
+    {
+        return _Name == _settings.ProfileDefaults().DefaultAppearance().ColorSchemeName();
+    }
+
+    void ColorSchemeViewModel::RefreshIsDefault()
+    {
+        _NotifyChanges(L"IsDefaultScheme");
+    }

```

Comment:
For the record, I hate this. But I think it's a necessary evil, and I tried to limit how exposed this is by making it only available through the impl type.

### Why do we need it?
- The "default" tag is visibly bound to the `IsDefaultScheme` getter above, which queries the model to know the answer.
- When the user clicks the "Set as default" button, we update the "base layer" (yes, I'm still calling it that haha) to have the new color scheme as the default. The key thing here is that the base layer stores the _name_ not the CS object. So we _need_ to query the settings model to answer the question "are we the default?". 
- So now we have a problem: the model updated, but the view model has no idea what's changed. We need the view model to refresh it's value for `IsDefaultScheme`.
- the solution: in `ColorSchemesPageViewModel`, we go through all the schemes and notify that `IsDefaultScheme` has changed.
- alternatively, why not store `IsDefaultScheme` as an observable bool: decided against this because this just makes constructing the list of scheme VMs nasty. This also means that we're not "pure"-ly knowing if we're the default scheme. We're just caching whatever the parent VM tells us, which just feels icky.

Reply:
I understand, and agree. You are seen. :D

Reply:
After reading the PR, I still agree! This is _fine_. We can eventually fix the settings model.

FWIW: the way to make this observable is to just call PropertyChangedHandlers with the name `IsDefaultScheme`. Less icky, and you could hook it up by chaining the propertychanged handler.

Reply:
Hey, how does this work if you go edit Base Layer directly? and select a different scheme?

Reply:
> Hey, how does this work if you go edit Base Layer directly? and select a different scheme?

Good catch, but surprisingly this works as expected! I guess XAML automatically handles it since the binding is in the markup.